### PR TITLE
fix(learn): check if input has only one type attribute

### LINKED ISSUE #47627: Cat Photo App Step 44 allows multiple type attributes
### Describe the Issue

Step 44 will pass if the new radio `input` is given multiple `type` attributes. 

### Affected Page

https://www.freecodecamp.org/learn/2022/responsive-web-design/learn-html-by-building-a-cat-photo-app/step-44

### Your code

This will pass:

```
<input type="radio" type="text"> Indoor
```

### Expected behavior

Step should fail if there is more than one `type` attribute, even if they all have the value `radio`.

### Screenshots

_No response_

### System

- Device: Desktop
- OS: Linux (Fedora 36)
- Browser: Firefox
- Version: 104.0.2

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Lozarnez**
Checklist:

- [ ] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [ ] I have tested these changes either locally on my machine, or GitPod.

Closes #47627 

Added a test to check if the input has more than one attribute called "type"

### CODE REVIEW

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804dc.md [hunk 1]
```diff
@@ -67,6 +67,16 @@ Although you have set the new `input` element's `type` attribute to `radio`, it
...
+Your new `input` element should have only one `type` attribute. Remove any extras.
+
+```js
+assert($('input')[0]
+  .getAttributesNames()
+  .filter(attr => attr === 'type')
+  .length === 1
+);
+```
+
```
Comment (jeremylt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804dc.md:79]: Should this come before checking what the specific value is?

**Review by jeremylt — DISMISSED** [review]
LGTM

**Comment by raisedadead** [comment]
Hi @Lozarnez 

Thanks for opening this PR.


Quick Tip:

You should not reference the original issue number multiple times, in commits, in PR titles, etc. 

One reference in the PR description (`Closes XXXXXX`) is plenty. Adding more makes it look ugly in the timelines etc. 

Now that this PR is open, you do not need to change the commit message; be mindful next time.

**Review by Sboonny — DISMISSED** [review]
Hi @Lozarnez, the required input `<input type="radio"> Indoor` fails.

image

can you update it, if you need help, join us in [our chat room](https://discord.gg/PRyKn3Vbay) or [the forum](https://forum.freecodecamp.org/c/contributors/3).

while we are updating the step, can we update the seed code to have an empty line for quality of life experience

so this code in the seed, can have an empty line for comfort

```diff
- --fcc-editable-region--
-         <input type="text" name="catphotourl" placeholder="cat photo URL" required>
- --fcc-editable-region--

+ --fcc-editable-region--
+ 
+        <input type="text" name="catphotourl" placeholder="cat photo URL" required>
+ --fcc-editable-region--
```

**Review by majestic-owl448 — DISMISSED** [review]
Please set up freeCodeCamp locally and test your code.

This would have failed.

**Comment by ieahleen** [comment]
Hey there, will you continue working on this PR?

File: curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804dc.md [hunk 2]
```diff
@@ -125,6 +135,7 @@ assert(
         <h2>Cat Form</h2>
         <form action="https://freecatphotoapp.com/submit-cat-photo">
 --fcc-editable-region--
+
```
Comment (jeremylt) [inline curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804dc.md:138]:
```suggestion
```

Comment (jeremylt) [reply curriculum/challenges/english/14-responsive-web-design-22/learn-html-by-building-a-cat-photo-app/5ef9b03c81a63668521804dc.md:138]: Woops, disreguard

**Review by jeremylt — APPROVED** [review]

**Review by raisedadead — APPROVED** [review]

**Pull request merged**